Sources/XCTest/Public/XCTAssert.swift

```diff
@@ -421,3 +423,14 @@ public func XCTAssertThrowsError<T>(_ expression: @autoclosure () throws -> T, _
         }
     }
 }
+
+public func XCTAssertNoThrow<T>(_ expression: @autoclosure () throws -> T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) {
+    _XCTEvaluateAssertion(.noThrow, message: message, file: file, line: line) {
+        do {
+             _ = try expression()
+            return .success
+        } catch _ {
+            return .expectedFailure("threw error")
```

Comment:
I think we're going to want to include a textual description of the error that was thrown here. For comparison, the overlay implementation looks like [this](https://github.com/apple/swift/blob/master/stdlib/public/SDK/XCTest/XCTest.swift#L1043).

Making this change will impact how you specify the expected failure message in the functional test suite, as @modocache was discussing in his comment.